apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.test.js

```diff
@@ -2,1133 +2,493 @@
+    const residentUser = await makeClientWithResidentUser()
+    const [resident] = await registerResidentByTestClient(residentUser, {
+        address: property.address,
+        addressMeta: property.addressMeta,
+        unitName: billingAccount.unitName,
+    })
+    residentUser.resident = resident
+    const [serviceConsumer] = await registerServiceConsumerByTestClient(residentUser, {
+        residentId: resident.id,
+        accountNumber: billingAccount.number,
+        organizationId: organization.id,
+    })
+    residentUser.serviceConsumer = serviceConsumer
+    return residentUser
+}
 
 describe('AllResidentBillingReceiptsService', () => {
+    const data = {
+        admin: null,
+        userClient: null,
+        supportClient: null,
+        integrationClient: null,
+        organization: null,
+        property: null,
+        billingContext: null,
+        acquiringContext: null,
+    }
+    beforeAll(async () => {
+        data.admin = await makeLoggedInAdminClient()
+        const { organization, property, ...userClient } =  await makeClientWithProperty()
+        data.userClient = userClient
+        data.organization = organization
+        data.property = property
+        data.supportClient = await makeClientWithSupportUser()
+        const [integration] = await createTestBillingIntegration(data.supportClient, { isTrustedBankAccountSource: true })
+        const [billingContext] = await createTestBillingIntegrationOrganizationContext(userClient, organization, integration, {
+            status: BILLING_CONTEXT_FINISHED_STATUS,
+        })
+        data.billingContext = billingContext
+        const [acquiringIntegration] = await createTestAcquiringIntegration(data.supportClient, {
+            canGroupReceipts: true,
+        })
+        const [acquiringContext] = await createTestAcquiringIntegrationContext(userClient, organization, acquiringIntegration, {
+            status: ACQUIRING_CONTEXT_FINISHED_STATUS,
+        })
+        data.acquiringContext = acquiringContext
+        data.integrationClient = await makeClientWithServiceUser()
+        await createTestBillingIntegrationAccessRight(data.supportClient, integration, data.integrationClient.user)
+    })
 
-    describe('fields for resident',  () => {
-        it('returns all required fields', async () => {
-            const userClient = await makeClientWithProperty()
-            const support = await makeClientWithSupportUser()
-
-            const [integration] = await createTestBillingIntegration(support)
-            const [billingContext] = await createTestBillingIntegrationOrganizationContext(userClient, userClient.organization, integration)
+    describe('Resident access and filters', () => {
 
-            const integrationClient = await makeClientWithServiceUser()
-            await createTestBillingIntegrationAccessRight(support, integration, integrationClient.user)
+        it('should contain fields for resident', async () => {
```

Comment:
Why `it`? Isn't it used for `spec.js` tests only?

Reply:
Done